src/librustdoc/html/render/print_item.rs

```diff
@@ -1571,6 +1573,38 @@ fn document_type_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
                     pl = if bytes == 1 { "" } else { "s" },
                 );
             }
+            if let Variants::Multiple { variants, .. } = &ty_layout.layout.variants {
+                if !variants.is_empty() {
+                    w.write_str(
+                        "<p>\
+                            <strong>Size for each variant:</strong>\
+                            <ul>",
+                    );
+
+                    let adt = if let Adt(adt, _) = ty_layout.ty.kind() {
+                        adt
+                    } else {
+                        bug!("not an adt")
+                    };
+
+                    for (index, layout) in variants.iter_enumerated() {
+                        let ident = adt.variants[index].ident;
+                        if layout.abi.is_unsized() {
+                            writeln!(w, "<li><code>{name}</code> (unsized)</li>", name = ident);
```

Comment:
This looks copied from the code above - do you mind separating into a function instead?